packages/components/psammead-topic-tags/src/index.jsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import { string, oneOf, shape, node } from 'prop-types';
+import styled from '@emotion/styled';
+import { C_LUNAR, C_EBON, C_METAL } from '@bbc/psammead-styles/colours';
+import { GEL_SPACING_HLF, GEL_SPACING } from '@bbc/gel-foundations/spacings';
+import { getSansRegular } from '@bbc/psammead-styles/font-styles';
+import { getBrevier } from '@bbc/gel-foundations/typography';
+import { scriptPropType } from '@bbc/gel-foundations/prop-types';
+
+const [TAG_TEXT_PAD_Y, TAG_TEXT_PAD_X, ROW_SPACING] = [
+  '0.375rem',
+  '0.4375rem',
+  '0.0625rem',
+];
+
+const topicTagsContainer = `
+  display: flex;
+  flex-wrap: wrap;
+  margin-top: -${ROW_SPACING};
+  margin-bottom: 0;
+  margin-left: -${GEL_SPACING_HLF};
+  margin-right: -${GEL_SPACING_HLF};
+  padding: 0;
+`;
+
+const SingleTopicTagContainer = styled.div`
+  ${topicTagsContainer}
+`;
+
+const TopicsList = styled.ul`
+  list-style-type: none;
+  ${topicTagsContainer}
+`;
+
+const SingleTopicTagItem = styled.div`
+  ${({ service }) => getSansRegular(service)}
+  ${({ script }) => script && getBrevier(script)}
+
+  word-break: break-word;
+  margin-top: ${ROW_SPACING};
+  margin-left: ${GEL_SPACING_HLF};
+  margin-right: ${GEL_SPACING_HLF};
+  a {
+    display: flex;
+    padding-top: ${GEL_SPACING};
+    padding-bottom: ${GEL_SPACING};
+    text-decoration: none;
+    color: ${C_EBON};
+
+    &:hover,
+    &:focus {
+      text-decoration: underline;
+    }
+    &:visited {
+      color: ${C_METAL};
+    }
+  }
+
+  a > span {
+    background-color: ${C_LUNAR};
+    padding: ${TAG_TEXT_PAD_Y} ${TAG_TEXT_PAD_X};
+  }
+`;
+
+export const TopicTag = ({ topicName, topicLink }) => (
+  <a href={topicLink}>
+    <span>{topicName}</span>
+  </a>
+);
+
+export const TopicTags = ({ dir, children, script, service }) => {
+  if (children.length === 0) return null;
+
+  const hasMultipleChildren = children.length > 1;
+
+  return (
+    <>
+      {hasMultipleChildren ? (
+        <TopicsList dir={dir} role="list" service={service} script={script}>
+          {children.map((child, i) => {
+            if (child.type !== TopicTag) return null;
+            const key = i;
```

Comment:
Is there a reason we are not using `i` directly? If it's for readability, we could instead rename `i` to `index`.

Reply:
Thanks for your comments and suggestions @simonsinclair, I've addressed most of them now 👍

Regarding this, the reason `key` is declared and `i` is used indirectly is so that the linter ignores the fact that array index is being used for the component key. We discussed that this isn't an issue since the `children` array being mapped on won't change order. However, I've just realised that I can disable the linter using the wonderful `//eslint-disable-next-line react/no-array-index-key` comment in this case. Do you think that's ok? Thanks!